### PR TITLE
[Torch-MLIR] add support for stable diffusion model

### PR DESCRIPTION
**Pull request by zzpmiracle**

### CODE REVIEW

File: pytorch_blade/pytorch_blade/torch-mlir/lib/Conversion/TorchToMhlo/DiscTorchToMhlo.cpp [hunk 1]
```diff
@@ -189,24 +189,10 @@ class ConvertAtenExtractOp : public OpConversionPattern<AtenOpT> {
...
-          rewriter.replaceOpWithNewOp<arith::TruncIOp>(op, outTy, output);
-        }
-      } else {
-        if (toWider) {
-          rewriter.replaceOpWithNewOp<arith::ExtFOp>(op, outTy, output);
-        } else {
-          rewriter.replaceOpWithNewOp<arith::TruncFOp>(op, outTy, output);
-        }
-      }
+      // outTy and elemTy may be diffrent datatypes like int to float
```
Comment (tanyokwok) [inline pytorch_blade/pytorch_blade/torch-mlir/lib/Conversion/TorchToMhlo/DiscTorchToMhlo.cpp:192]: I think `mhlo::ConvertOp` may introduce much redundant data access here. How about extracting the element first and then handling the type conversion/casting carefully as before？

File: pytorch_blade/pytorch_blade/compiler/jit/torch/shape_analysis.cpp [hunk 2]
```diff
@@ -1027,6 +1014,20 @@ class ShapePropagator : public PropertyPropBase {
             "aten::narrow(Tensor self, int dim, int start, int length) -> Tensor",
             "aten::alias(Tensor self) -> Tensor",
             "aten::zero_(Tensor self) -> Tensor",
+#if PYTORCH_MAJOR_VERSION == 1 && PYTORCH_MINOR_VERSION > 6
```
Comment (Yancey1989) [inline pytorch_blade/pytorch_blade/compiler/jit/torch/shape_analysis.cpp:1017]: PR #707 added a macro `PYTORCH_VERSION_GE ` to check the PyTorch version, which is more clearly.

**Review by Yancey1989 — COMMENTED** [review]
better to add some unit tests .

**Comment by qiuxiafei** [comment]
This PR doesn't have a good name ...

File: pytorch_blade/pytorch_blade/torch-mlir/lib/Conversion/TorchToMhlo/DiscTorchToMhlo.cpp [hunk 3]
```diff
@@ -184,33 +184,156 @@ class ConvertAtenExtractOp : public OpConversionPattern<AtenOpT> {
...
+        return success();
       }
-    } else {
-      rewriter.replaceOpWithNewOp<tensor::ExtractOp>(op, outTy, input);
     }
-    return success();
+
+    // integer to float
+    // 1-bit integers need to be treated as signless.
+    if (elemTy.isInteger(1)) {
```
Comment (tanyokwok) [inline pytorch_blade/pytorch_blade/torch-mlir/lib/Conversion/TorchToMhlo/DiscTorchToMhlo.cpp:218]: Can we create file checks for all those conversion patterns?

**Comment by zzpmiracle** [comment]
Split in #867 . Closed.

**Pull request closed without merging**